### PR TITLE
add svg basics test and example

### PR DESCRIPTION
**Pull request by sepiropht**
[#448](https://github.com/rustwasm/wasm-bindgen/issues/448)

### CODE REVIEW

**Comment by fitzgen** [comment]
Looks like there's a bunch of compilation errors: https://travis-ci.org/rustwasm/wasm-bindgen/jobs/420500433#L3876

Does this compile and run OK for you locally?

**Comment by sepiropht** [comment]
In fact locally my example doesn't build but neither the old one (dom, hello-word...etc)
I used the script ./build.sh or i tried manually with wasm-bindgen cli i keep getting this errors.
`svg git:feature_svg_basics_exaple_testing ❯ ./build.sh     ⏎ ✹
+ cargo +nightly build --target wasm32-unknown-unknown
error: failed to resolve patches for `https://github.com/rust-lang/crates.io-index`
Caused by:
  patch for `wasm-bindgen` in `https://github.com/rust-lang/crates.io-index` did not resolve to any crates. If this is unexpected, you may wish to consult: https://github.com/rust-lang/cargo/issues/4678`
This why i push to see the build result in travis, sorry for that.

**Comment by sepiropht** [comment]
I get this errors: `Module not found: Error: Can't resolve './svg' in '/home/travis/build/rustwasm/wasm-bindgen/examples/svg'`
It seems to me that i followed exactly the same structure that the others examples, but i'm certainly missing something, as it is not working.

**Comment by fitzgen** [comment]
Have you tried a `cargo update` perhaps?

**Comment by sepiropht** [comment]
Thanks @fitzgen `cargo update` solved all issues !

File: examples/svg/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+pub fn run() {
+    let svg = document.createElementNS("http://www.w3.org/2000/svg", "svg"); 
+    svg.set_attribute("width", &JsValue::from(200));
+    svg.set_attribute("height", &JsValue::from(100));
+    document.body().append_child(&svg);
+
+    let path = document.createElementNS("http://www.w3.org/2000/svg", "path");
+    path.set_attribute("d", &JsValue::from("M100,0 L200,100 100,200 0,100Z"));
+    path.set_attribute("fill", &JsValue::from("red"));    
+    svg.append_child(&path);
```
Comment (fitzgen) [inline examples/svg/src/lib.rs]: I think we want something that is slightly more compelling than a red diamond (I think that's what this is?)

How about porting

```svg
  <svg width="100%" height="100%">
    <rect width="100%" height="100%" fill="red" />
    <circle cx="100%" cy="100%" r="150" fill="blue" stroke="black" />
    <polygon points="120,0 240,225 0,225" fill="green"/>
    <text x="50" y="100" font-family="Verdana" font-size="55"
          fill="white" stroke="black" stroke-width="2">
            Hello!
    </text>
  </svg>
```

from https://developer.mozilla.org/en-US/docs/Learn/HTML/Multimedia_and_embedding/Adding_vector_graphics_to_the_Web to `web-sys`?

File: examples/svg/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+// These definitions need to be hand-written today but the current vision is
+// that we'll use WebIDL to generate this `extern` block into a crate which you
+// can link and import. There's a tracking issue for this at
+// https://github.com/rustwasm/wasm-bindgen/issues/42
+//
+// In the meantime these are written out by hand and correspond to the names and
+// signatures documented on MDN, for example
+#[wasm_bindgen]
+extern "C" {
+    type HTMLDocument;
```
[the remark this answers is not in the document]
Comment (sepiropht) [reply examples/svg/src/lib.rs]: Yes i understand that, but even in raw javascript we don't use any svg class to create and append svg in the dom.

[the remark this answers is not in the document]
Comment (fitzgen) [reply examples/svg/src/lib.rs]: Sorry I wasn't clear: it should be using `web_sys::HtmlDocument` isntead of a custom import, and `web_sys::HtmlDocument::create_element_ns` etc..

**Comment by xliiv** [comment]
What's the status for this?

**Comment by alexcrichton** [comment]
I'm gonna close this as it's been a bit quiet and stale for some time now, but we can of course resubmit at any time!

**Pull request closed without merging**